packages/react-instantsearch/src/components/RefinementList.js

```diff
@@ -18,32 +25,44 @@ class RefinementList extends Component {
       isRefined: PropTypes.bool.isRequired,
     })),
     showMore: PropTypes.bool,
+    isSearchable: PropTypes.bool,
+    isFromSearch: PropTypes.bool.isRequired,
     limitMin: PropTypes.number,
     limitMax: PropTypes.number,
   };
 
-  renderItem = item =>
+  selectItem = item => {
+    this.props.refine(item.value);
+  };
+
+  renderItem = item => {
+    const label = item._highlightResult
+      ? <Highlight attributeName="label" hit={item}/>
+      : item.label;
+
+    return (
       <label>
         <input
           {...cx('itemCheckbox', item.isRefined && 'itemCheckboxSelected')}
           type="checkbox"
           checked={item.isRefined}
-          onChange={() => this.props.refine(item.value)}
+          onChange={() => this.selectItem(item)}
         />
         <span {...cx('itemBox', 'itemBox', item.isRefined && 'itemBoxSelected')}></span>
         <span {...cx('itemLabel', 'itemLabel', item.isRefined && 'itemLabelSelected')}>
-          {item.label}
+          {label}
         </span>
         {' '}
         <span {...cx('itemCount', item.isRefined && 'itemCountSelected')}>
           {item.count}
         </span>
-      </label>
-    ;
+      </label>);
+  };
 
   render() {
-    return (
-      <List
+    const sortedFacets = orderBy(this.props.items, ['isRefined', 'count', 'name'], ['desc', 'desc', 'asc']);
```

Comment:
The results of search for facet values is a "search" which means that the order defined is computed by the engine, as it would for a normal search. Therefore I wouldn't sort them again.